### PR TITLE
support tomcat  dubbo and httpclient plugins

### PR DESCRIPTION
**Pull request by ascrutae**
Example:
Tomcat  --> Dubbo

Here is the trace segment of  tomcat 
```
{
    "traceSegmentId": "S.1499011869343.-2070780384.32757.70.1",
    "startTime": 1499011869343,
    "endTime": 1499011871919,
    "spans": [
        {
            "stackDepth": 0,
            "spanId": 1,
            "parentSpanId": 0,
            "tags": [
                {
                    "key": "url",
                    "value": "rest://192.168.1.8:20880/com.a.eye.skywalking.test.persistence.PersistenceService.query(String)"
                }
            ],
            "operationName": "com.a.eye.skywalking.test.persistence.PersistenceService.query(String)",
            "operationId": 0,
            "layer": "RPC_FRAMEWORK",
            "startTime": 1499011869350,
            "endTime": 1499011871896,
            "errorOccurred": false,
            "componentId": 3
        },
        {
            "stackDepth": 0,
            "currentMaxDepth": 1,
            "spanId": 1,
            "parentSpanId": 0,
            "tags": [
                {
                    "key": "url",
                    "value": "http://localhost:48080/portal/"
                },
                {
                    "http.method": "GET"
                }
            ],
            "operationName": "/portal/",
            "operationId": 0,
            "layer": "HTTP",
            "startTime": 1499011869343,
            "endTime": 1499011871919,
            "errorOccurred": false,
            "componentId": 1
        }
    ],
    "applicationId": 1,
    "relatedGlobalTraces": {
        "relatedGlobalTraces": [
            {
                "id": "T.1499011824847.-2070780384.32757.69.2"
            }
        ]
    },
    "ignore": false
}
```

Here is the trace segment of dubbo server side
```
{
    "traceSegmentId": "S.1499011826245.747089479.32752.69.1",
    "startTime": 1499011826244,
    "endTime": 1499011837840,
    "refs": [
        {
            "traceSegmentId": "S.1499011824845.-2070780384.32757.69.1",
            "spanId": 1,
            "applicationId": 1,
            "peerHost": "192.168.1.8:20880",
            "peerId": 0,
            "operationName": "/portal/",
            "operationId": 0,
            "distributedTraceIds": [
                {
                    "id": "T.1499011824847.-2070780384.32757.69.2"
                }
            ]
        }
    ],
    "spans": [
        {
            "stackDepth": 0,
            "currentMaxDepth": 2,
            "spanId": 0,
            "parentSpanId": -1,
            "tags": [
                {
                    "key": "url",
                    "value": "rest://192.168.1.8:20880/com.a.eye.skywalking.test.persistence.PersistenceService.query(String)"
                }
            ],
            "operationName": "com.a.eye.skywalking.test.persistence.PersistenceService.query(String)",
            "operationId": 0,
            "layer": "RPC_FRAMEWORK",
            "startTime": 1499011831227,
            "endTime": 1499011837840,
            "errorOccurred": false,
            "componentId": 3
        }
    ],
    "applicationId": 3,
    "relatedGlobalTraces": {
        "relatedGlobalTraces": [
            {
                "id": "T.1499011824847.-2070780384.32757.69.2"
            }
        ]
    },
    "ignore": false
}
```

### CODE REVIEW

File: apm-sniffer/apm-sdk-plugin/dubbo-plugin/src/main/java/org/skywalking/apm/plugin/dubbo/DubboInterceptor.java [hunk 1]
```diff
@@ -39,26 +40,23 @@
...
-        Tags.COMPONENT.set(span, DUBBO_COMPONENT);
-        Tags.SPAN_LAYER.asRPCFramework(span);
-        span.setPeerHost(requestURL.getHost());
-        span.setPort(requestURL.getPort());
+        AbstractSpan span;
 
+        final String host = requestURL.getHost();
+        final int port = requestURL.getPort();
+        objInst.setSkyWalkingDynamicField("aa");
+        objInst.getSkyWalkingDynamicField();
```
Comment (wu-sheng) [inline apm-sniffer/apm-sdk-plugin/dubbo-plugin/src/main/java/org/skywalking/apm/plugin/dubbo/DubboInterceptor.java]: These two lines should be removed.

File: apm-sniffer/apm-sdk-plugin/httpClient-4.x-plugin/src/main/java/org/skywalking/apm/plugin/httpClient/v4/HttpClientExecuteInterceptor.java [hunk 2]
```diff
@@ -12,68 +12,47 @@
...
+        final ContextCarrier contextCarrier = new ContextCarrier();
         AbstractSpan span;
+        String remotePeer = httpHost.getHostName() + ":" + httpHost.getPort();
         try {
             URL url = new URL(httpRequest.getRequestLine().getUri());
-            span = ContextManager.createSpan(url.getPath());
+            span = ContextManager.createExitSpan(url.getPath(), contextCarrier, remotePeer);
         } catch (MalformedURLException e) {
-            span = ContextManager.createSpan(httpRequest.getRequestLine().getUri());
+            span = ContextManager.createExitSpan(httpRequest.getRequestLine().getUri(), contextCarrier, remotePeer);
```
Comment (wu-sheng) [inline apm-sniffer/apm-sdk-plugin/httpClient-4.x-plugin/src/main/java/org/skywalking/apm/plugin/httpClient/v4/HttpClientExecuteInterceptor.java]: Why do the nearly same thing in `catch` block? What is the difference between URL and URI?

File: apm-sniffer/apm-sdk-plugin/dubbo-plugin/src/main/java/org/skywalking/apm/plugin/dubbo/DubboInterceptor.java (hunk 1 above)
Comment (ascrutae) [inline apm-sniffer/apm-sdk-plugin/dubbo-plugin/src/main/java/org/skywalking/apm/plugin/dubbo/DubboInterceptor.java]: OK，

**Comment by ascrutae** [comment]
I had finished it .

**Pull request merged**